Assignment2/InviteFriendsTest.cpp

```diff
@@ -0,0 +1,74 @@
+#include "InviteFriends.h"
+#include "InviteFriendsTest.h"
+
+using namespace std;
+CPPUNIT_TEST_SUITE_REGISTRATION ( InviteFriendsTest );
+
+void InviteFriendsTest::testInviteFriends(void)
+{
+    InviteFriends inviteFriends;
+
+    string fileName = "friends1.json";
+
+    if ( !inviteFriends.readFile(fileName) )
+    {
+        cout << "Cannot read the file\n\n";
+    }
+
+    vector < pair < int, string > > friendsComingToParty = inviteFriends.getIdsWithName();
+
+    for ( int i = 0; i < friendsComingToParty.size(); ++i )
+    {
+        cout << friendsComingToParty[i].second << "\n\n";
+    }
+
+    fileName = "friends2.json";
+
+    bool failed = false;
```

Comment:
these tests are very weak - they only test readFile works..no assertions around the actual friends being output. cout at line 22 should be been replaced by assertions.

Reply:
We are not supposed to allow negative coordinates for this specific case right ?  because radius of 100 km around banglore does not include negative coordinates. Same doubt for North/South Poles.

Reply:
a well written code should be agnostic to whether it is bangalore or any other lat/lng

Reply:
Yes, I agree with you but I am a bit confused here because as far as I can see the latitude/longitude can be negative also. So, why would be avoid the negative latitude/longitude ?

Reply:
we should not avoid negative lat/lng as they are valid inputs - that is what I meant by well written code